vis/optimizer.py

```diff
@@ -105,82 +113,130 @@ def _get_seed_input(self, seed_input):
-                channels_first` or `(samples, image_dims..., channels)` if `image_data_format=channels_last`.
+            seed_inputs: A numpy array or list of numpy array.
+                The shape of an N-dim numpy array is `(samples, channels, image_dims...)` if `image_data_format=
+                channels_first`, Or it's `(samples, image_dims..., channels)` if `image_data_format=channels_last`.
                 Seeded with random noise if set to None. (Default value = None)
             max_iter: The maximum number of gradient descent iterations. (Default value = 200)
-            input_modifiers: A list of [InputModifier](vis.input_modifiers#inputmodifier) instances specifying
+            input_modifiers: A list of InputModifier, or a list of tuple of target input tensor and InputModifier.
+                [InputModifier](vis.input_modifiers#inputmodifier) instances specifying
                 how to make `pre` and `post` changes to the optimized input during the optimization process.
                 `pre` is applied in list order while `post` is applied in reverse order. For example,
                 `input_modifiers = [f, g]` means that `pre_input = g(f(inp))` and `post_input = f(g(inp))`
             grad_modifier: gradient modifier to use. See [grad_modifiers](vis.grad_modifiers.md). If you don't
                 specify anything, gradients are unchanged. (Default value = None)
-            callbacks: A list of [OptimizerCallback](vis.callbacks#optimizercallback) instances to trigger.
+            callbacks: A list of of OptimizerCallback, or a list of tuple of target tensor and OptimizerCallback.
+                [OptimizerCallback](vis.callbacks#optimizercallback) instances to trigger.
+                target tensor is usualy wrt_tensor, but when wrt_tensors is None, it is input_tensors.
+                That is, When optimizer is used ActivationMaximization.
             verbose: Logs individual losses at the end of every gradient descent iteration.
                 Very useful to estimate loss weight factor(s). (Default value = True)
 
         Returns:
             The tuple of `(optimized input, grads with respect to wrt, wrt_value)` after gradient descent iterations.
         """
-        seed_input = self._get_seed_input(seed_input)
+        seed_inputs = utils.listify(seed_inputs)
+        # Make seed_inputs as same length as input_tensors
+        for _ in range(len(seed_inputs), len(self.input_tensors)):
+            seed_inputs.append(None)
+        for i in range(len(seed_inputs)):
+            seed_inputs[i] = self._get_seed_input(i, seed_inputs[i])
+
         input_modifiers = input_modifiers or []
-        grad_modifier = _identity if grad_modifier is None else get(grad_modifier)
+        grad_modifier = (lambda x: x) if grad_modifier is None else get(grad_modifier)
 
         callbacks = callbacks or []
         if verbose:
-            callbacks.append(_PRINT_CALLBACK)
+            callbacks.append(Print())
```

Comment:
This is because the callback is stateless. We save on object creation overhead and memory. Thats why i made it global